opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -609,15 +609,18 @@ def __repr__(self):
     def _from_env_if_absent(
         cls, value: Optional[int], env_var: str, default: Optional[int] = None
     ) -> Optional[int]:
-        if value is cls.UNSET:
+        if value == cls.UNSET:
```

Comment:
Maybe I'm not fully understanding this scenario. When will `value` ever be "unset"? Isn't value supposed to be an `int`?

Reply:
The API usage looks like the following in order to not set a limit:

```python
SpanLimits(max_attributes=SpanLimits.UNSET)
```

SpanLimits.UNSET is a symbol that users can pass as a limit to set the limit to infinity.  We can freely change its internal value to anything without breaking any contracts. 

The reason why I used "unset" was that I thought -1 might be confusing as the spec disallows it for env vars. Internally we can still use -1 as the symbol value.

Reply:
Changed it back to -1 as this is probably causing more confusion and does not respect the function contract. I wonder why mypy didn't catch it.